activation/handler_test.go

```diff
@@ -1078,50 +1078,105 @@ func BenchmarkGetAtxHeaderWithConcurrentProcessAtx(b *testing.B) {
-	atx2.PrevATXID = types.ATXID{2, 3, 4} // should be fetched
-	atxHdlr.mockFetch.EXPECT().GetAtxs(gomock.Any(), []types.ATXID{atx2.PrevATXID}).Return(nil)
-	require.NoError(t, atxHdlr.FetchAtxReferences(context.Background(), atx2))
-
-	atx3 := newAtx(t, sig, challenge, nipost, 2, coinbase)
-	atx3.PositioningATX = types.EmptyATXID // should *NOT* be fetched
-	atx3.PrevATXID = types.ATXID{3, 4, 5}  // should be fetched
-	atxHdlr.mockFetch.EXPECT().GetAtxs(gomock.Any(), []types.ATXID{atx3.PrevATXID}).Return(nil)
-	require.NoError(t, atxHdlr.FetchAtxReferences(context.Background(), atx3))
-
-	atx4 := newAtx(t, sig, challenge, nipost, 2, coinbase)
-	atx4.PositioningATX = types.ATXID{5, 6, 7} // should be fetched
-	atx4.PrevATXID = types.EmptyATXID          // should *NOT* be fetched
-	atxHdlr.mockFetch.EXPECT().GetAtxs(gomock.Any(), []types.ATXID{atx4.PositioningATX}).Return(nil)
-	require.NoError(t, atxHdlr.FetchAtxReferences(context.Background(), atx4))
-
-	atx5 := newAtx(t, sig, challenge, nipost, 2, coinbase)
-	atx5.PositioningATX = types.EmptyATXID // should *NOT* be fetched
-	atx5.PrevATXID = types.EmptyATXID      // should *NOT* be fetched
-	require.NoError(t, atxHdlr.FetchAtxReferences(context.Background(), atx5))
-
-	atx6 := newAtx(t, sig, challenge, nipost, 2, coinbase)
-	atxid := types.ATXID{1, 2, 3}
-	atx6.PositioningATX = atxid // should be fetched
-	atx6.PrevATXID = atxid      // should be fetched
-	atxHdlr.mockFetch.EXPECT().GetAtxs(gomock.Any(), []types.ATXID{atxid}).Return(nil)
-	require.NoError(t, atxHdlr.FetchAtxReferences(context.Background(), atx6))
+	posATX := types.ATXID{1, 2, 3}
+	prevATX := types.ATXID{4, 5, 6}
+	commitATX := types.ATXID{7, 8, 9}
+
+	t.Run("valid prev and pos ATX", func(t *testing.T) {
+		t.Parallel()
+		atxHdlr := newTestHandler(t, goldenATXID)
+
+		coinbase := types.Address{2, 4, 5}
+		challenge := newChallenge(1, prevATX, posATX, types.LayerID(22).GetEpoch(), nil)
+		nipost := newNIPostWithChallenge(t, types.HexToHash32("55555"), []byte("66666"))
+		atx := newAtx(t, sig, challenge, nipost, 2, coinbase)
+
+		atxHdlr.mockFetch.EXPECT().GetAtxs(gomock.Any(), gomock.InAnyOrder([]types.ATXID{atx.PositioningATX, atx.PrevATXID})).Return(nil)
+		require.NoError(t, atxHdlr.FetchAtxReferences(context.Background(), atx))
+	})
+
+	t.Run("valid prev ATX and golden pos ATX", func(t *testing.T) {
+		t.Parallel()
+		atxHdlr := newTestHandler(t, goldenATXID)
+
+		coinbase := types.Address{2, 4, 5}
+		challenge := newChallenge(1, prevATX, goldenATXID, types.LayerID(22).GetEpoch(), nil)
+		nipost := newNIPostWithChallenge(t, types.HexToHash32("55555"), []byte("66666"))
+		atx := newAtx(t, sig, challenge, nipost, 2, coinbase)
+
+		atxHdlr.mockFetch.EXPECT().GetAtxs(gomock.Any(), []types.ATXID{atx.PrevATXID}).Return(nil)
+		require.NoError(t, atxHdlr.FetchAtxReferences(context.Background(), atx))
+	})
+
+	t.Run("valid prev ATX and empty pos ATX", func(t *testing.T) {
```

Comment:
That is correct, but when `FetchAtxReferences` is called we haven't checked yet for syntactically valid ATXs. So a node can send an ATX without a pos ATX, because it is only checked later if this is even valid or not.